mqtt/mqtt-bridge/src/pump/messages.rs

```diff
@@ -98,28 +98,25 @@ where
 
                     for sub in added {
                         let subscribe_to = sub.subscribe_to();
-                        let subscribe_result = self
-                            .subscription_handle
-                            .subscribe(SubscribeTo {
-                                topic_filter: subscribe_to.clone(),
-                                qos: QoS::AtLeastOnce, // TODO: get from config
-                            })
-                            .await;
 
-                        match subscribe_result {
-                            Ok(_) => match sub.to_owned().try_into() {
-                                Ok(mapper) => {
-                                    self.topic_mappers_updates.insert(&subscribe_to, mapper);
-                                }
-                                Err(e) => {
-                                    error!("topic rule could not be parsed {}. {}", subscribe_to, e)
+                        match sub.to_owned().try_into() {
+                            Ok(mapper) => {
+                                self.topic_mappers_updates.insert(&subscribe_to, mapper);
```

Comment:
should we insert only when sub succeeded?

Reply:
it means that the subscribe event was sent to the channel, but was still it doesn't mean it was sent to server